boringtun/src/device/mod.rs

```diff
@@ -1040,23 +1053,29 @@ impl Device {
                                     );
                                 }
                             } else if let Some(addr @ SocketAddr::V4(_)) = endpoint.addr {
-                                udp4.sendto(packet, addr);
-                                tracing::trace!(
-                                    message = "Writing packet to network v4",
-                                    interface = ?t.iface.name(),
-                                    packet_length = packet.len(),
-                                    src_addr = ?addr,
-                                    public_key = public_key
-                                );
+                                if let Err(err) = udp4.sendto(packet, addr) {
+                                    tracing::warn!(message = "Failed to write packet to network v4", error = ?err);
+                                } else {
+                                    tracing::trace!(
+                                        message = "Writing packet to network v4",
+                                        interface = ?t.iface.name(),
+                                        packet_length = packet.len(),
+                                        src_addr = ?addr,
+                                        public_key = public_key
+                                    );
+                                }
                             } else if let Some(addr @ SocketAddr::V6(_)) = endpoint.addr {
-                                udp6.sendto(packet, addr);
-                                tracing::trace!(
-                                    message = "Writing packet to network v6",
-                                    interface = ?t.iface.name(),
-                                    packet_length = packet.len(),
-                                    src_addr = ?addr,
-                                    public_key = public_key
-                                );
+                                if let Err(err) = udp6.sendto(packet, addr) {
+                                    tracing::warn!(message = "Failed to write packet to network v6", error = ?err);
+                                }else {
```

Comment:
formatting seems to be broken

Reply:
Yeah, and I don't know why. It seems rustfmt doesn't work on this part for some reason.
I'll fix it manually

Reply:
Fixed